v2/pkg/engine/resolve/loader.go

```diff
@@ -182,13 +194,22 @@ func (l *Loader) resolveAndMergeFetch(fetch Fetch, items []int) error {
 	switch f := fetch.(type) {
 	case *SingleFetch:
 		res := &result{
-			out: pool.BytesBuffer.Get(),
+			out:         pool.BytesBuffer.Get(),
+			loadContext: context.Background(),
```

Comment:
Should this wrap the engine context? (so that the engine can cancel the load call)